### DISCUSSION TITLE
Can PyG do temporal graph classification?

### DISCUSSION THREAD

**Discussion by johnnytam100** [pyg-team/pytorch_geometric#4291] [Q&A] [answered]
I'm a beginner of using PyG and the task may be a bit advanced, but **temporal graph classification** seems to be a solution to a task in my field (bioinformatics).

May I know if PyG can do temporal graph classification for the current time point?

i.e. multiple temporal (dynamic) graphs with each associated with a label, and I want to ingest them into PyG to classify each temporal graph.

I wonder if a code example/ snippet/ tutorial on this task is available.

Thanks in advance. 👍

**Comment by rusty1s** [comment, marked as answer]
We are currently working on better temporal graph support via `data.TemporalData` and `loader.TemporalDataLoader`. Currently, we provide only a single temporal GNN model: the Temporal Graph Network (`TGN`), see [here](https://github.com/pyg-team/pytorch_geometric/blob/master/examples/tgn.py) for an example.

It might be also worth it to check out [PyTorch Geometric Temporal](https://github.com/benedekrozemberczki/pytorch_geometric_temporal) :)

**Reply by johnnytam100** [reply]
thank you @rusty1s !
